app/models/product_import/product_importer.rb

```diff
@@ -182,7 +182,7 @@ def init_product_importer
       @spreadsheet_data = SpreadsheetData.new(@entries, @import_settings)
       @validator = EntryValidator.new(@current_user, @import_time, @spreadsheet_data,
                                       @editable_enterprises, @inventory_permissions, @reset_counts,
-                                      @import_settings)
+                                      @import_settings, build_all_entries)
```

Comment:
if I understand correctly, this way we are loading the full spreadsheet all the time so that we defend from these unit_types problems. I dont think there's any other way, but this will increase the import time, right? maybe users/testers should be warned?

Reply:
Yes, you're right. It would increase import time, but fortunately nothing that would touch the database. We don't validate the items in this collection. I'll note this in the testing and release notes.

This mostly sets and reads the attributes. It would use more memory, but I guess only an MB or two even for larger import files (sample CSV file in #3843 with 252 rows has 76,955 characters). :sweat_smile:

Reply:
:crossed_fingers: Hoping the difference is negligible.

@luisramos0 I updated the testing and release notes, FYI.